<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Prioritization of functional rare genetic variants by integrating
    genomic annotations and RNA-seq â€¢ WatershedR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Prioritization of functional rare genetic variants by integrating
    genomic annotations and RNA-seq">
<meta property="og:description" content="Watershed is an unsupervised probabilistic framework that
    integrates genomic annotations and RNA-seq outlier calls to identify
    the probability a rare variant has a functional effect on a particular
    RNA-seq outlier phenotype (examples of outlier phenotypes can be, but
    are not limited to total expression, splicing, or ASE). Watershed
    extends our previous model RIVER (which can also be run via this
    package) by incoorperating information from multiple outlier
    phenoytpes into one model, where predictions for functional effects in
    one outlier phenotype are informed by observed outlier calls in
    another phenotype.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">WatershedR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="articles/WatershedR.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/nicolerg/WatershedR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="watershed">Watershed<a class="anchor" aria-label="anchor" href="#watershed"></a>
</h1></div>
<!-- badges: start -->

<hr>
<p><strong>This repository is an R-package implementation of the original command-line scripts available at the <a href="https://github.com/BennyStrobes/Watershed" class="external-link">upstream repository</a> to streamline installation and dependency management.</strong></p>
<hr>
<p>Watershed is an unsupervised probabilistic framework that integrates genomic annotations and RNA-seq outlier calls to identify the probability a rare variant has a functional effect on a particular RNA-seq outlier phenotype (examples of outlier phenotypes can be, but are not limited to total expression, splicing, or ASE). Watershed extends our previous model <a href="https://www.nature.com/articles/nature24267" class="external-link">RIVER</a> (which can also be run via this package) by incorporating information from multiple outlier phenotypes into one model, where predictions for functional effects in one outlier phenotype are informed by observed outlier calls in another phenotype. Please see <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7646251/" class="external-link">our publication in <em>Science</em></a> for more details.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>Install this R package from the GitHub repository:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://devtools.r-lib.org/" class="external-link">"devtools"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"nicolerg/WatershedR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="input-data">Input data<a class="anchor" aria-label="anchor" href="#input-data"></a>
</h2>
<p>For details about the input file format, see the <a href="https://nicolerg.github.io/WatershedR/reference/evaluate_watershed.html" class="external-link">docs</a>:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/nicolerg/WatershedR" class="external-link">WatershedR</a></span><span class="op">)</span></span>
<span><span class="op">?</span><span class="va">evaluate_watershed</span></span></code></pre></div>
<p>An example input file with 18 genomic annotations and 3 outlier p-values can be found in <a href="example_data/watershed_example_data.txt">example_data/watershed_example_data.txt</a>.</p>
<p>Another example input file with 18 genomic annotations and 1 outlier p-value can be found in <a href="example_data/river_example_data_pheno_1.txt">example_data/river_example_data_pheno_1.txt</a>.</p>
</div>
<div class="section level2">
<h2 id="running-watershed">Running Watershed<a class="anchor" aria-label="anchor" href="#running-watershed"></a>
</h2>
<p>This package provides two functions useful to users looking to apply Watershed to their data:</p>
<ol style="list-style-type: decimal">
<li><p><strong>evaluate_watershed():</strong> This function is used to train a Watershed model on non-N2 pairs and evaluate model on held-out N2-pairs. This allows the user to get an idea of the accuracy of Watershed applied to their data.</p></li>
<li><p><strong>predict_watershed():</strong> This function trains a Watershed model on training data and predicts Watershed posterior probabilities (using Watershed parameters optimized in training) on all gene-individual in a much larger prediction data set.</p></li>
</ol>
<p>Both of these functions can be run with three different models:</p>
<ul>
<li><p><strong>Watershed_exact:</strong> Watershed where parameters are optimized via exact inference (tractable and recommended when the number of dimensions (<code>E</code>) is small. A general rule of thumb is if the number of dimensions (<code>E</code>) is less than equal to 4, exact inference should be used).</p></li>
<li><p><strong>Watershed_approximate:</strong> Watershed where parameters are optimized using approximate inference. This approach is tractable when the number of dimensions (<code>E</code>) is large. For example, we used this to model the related outlier signals from 49 tissues (see <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7646251/" class="external-link">our publication</a>).</p></li>
<li><p><strong>RIVER:</strong> A <a href="https://www.nature.com/articles/nature24267" class="external-link">previously published method</a>. Used if the number of dimensions (<code>E</code>) is 1.</p></li>
</ul>
<p>See the <a href="https://nicolerg.github.io/WatershedR/reference/index.html" class="external-link">function documentation</a> for details about the outputs. See the <a href="https://nicolerg.github.io/WatershedR/articles/WatershedR.html" class="external-link">Get Started vignette</a> for examples of how to run these functions.</p>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>If you use this R package, please site our publication:</p>
<p>N.M. Ferraro, B.J. Strober, J. Einson, N.S. Abell, F. Aguet, A.N. Barbeira, M. Brandt, M. Bucan, S.E. Castel, J.R. Davis, et al., TOPMed Lipids Working Group, GTEx Consortium, Transcriptomic signatures across human tissues identify functional rare genetic variation. Science 369, eaaz5900 (2020).</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/nicolerg/WatershedR/" class="external-link">Browse source code</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing WatershedR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Ben Strober <br><small class="roles"> Author </small>  </li>
<li>Nicole Gay <br><small class="roles"> Maintainer, contributor </small>  </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/nicolerg/WatershedR/actions/workflows/R-CMD-check.yaml" class="external-link"><img src="https://github.com/nicolerg/WatershedR/actions/workflows/R-CMD-check.yaml/badge.svg" alt="R-CMD-check"></a></li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Ben Strober, Nicole Gay.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
